assets/javascripts/discourse/models/user-theme.js

```diff
@@ -1,3 +1,14 @@
 import Theme from "admin/models/theme";
+import UserThemeSettings from "./user-theme-settings";
 
-export default class UserTheme extends Theme {}
+export default class UserTheme extends Theme {
+  static munge(json) {
+    if (json.settings) {
+      json.settings = json.settings.map((setting) =>
+        UserThemeSettings.create(setting)
```

Comment:
We override this to be able to pass our own UserThemeSettings which override loadMetadata